src/Sdk/Cerbos/Sdk/CheckResult.cs

```diff
@@ -17,51 +21,118 @@ namespace Cerbos.Sdk
         /// </summary>
         public bool IsAllowed(string action)
         {
-            return _effects[action].Equals(Effect.Allow);
+            if (_result == null)
+            {
+                return false;
+            }
+
+            return _result.Actions.ToDictionary(
+                x => x.Key,
+                x => x.Value
+            ).TryGetValue(action, out var effect) && effect == Effect.Allow;
         }
 
         public Dictionary<string, bool> GetAll()
         {
-            var all = new Dictionary<string, bool>();
-            foreach(var effect in _effects)
+            return _result.Actions.ToDictionary(
+                x => x.Key,
+                x => x.Value == Effect.Allow
+            );
+        }
+
+        public Types.Meta GetMeta()
+        {
+            return new Types.Meta(_result.Meta);
+        }
+        
+        public Types.Outputs GetOutputs()
+        {
+            return new Types.Outputs(_result.Outputs.ToList());
+        }
+
+        public List<ValidationError> GetValidationErrors()
+        {
+            return _result.ValidationErrors.ToList();
+        }
+
+        public CheckResourcesResponse.Types.ResultEntry GetRaw()
+        {
+            return _result;
+        }
+
+        public static class Types 
+        {
+            public sealed class Meta
             {
-                all[effect.Key] = effect.Value.Equals(Effect.Allow);
+                private readonly CheckResourcesResponse.Types.ResultEntry.Types.Meta _meta;
+                public Dictionary<string, CheckResourcesResponse.Types.ResultEntry.Types.Meta.Types.EffectMeta> Actions => GetActions();
+                public List<string> EffectiveDerivedRoles => GetEffectiveDerivedRoles();
+
+                internal Meta(CheckResourcesResponse.Types.ResultEntry.Types.Meta meta)
+                {
+                    _meta = meta;
+                }
+            
+                public List<string> GetEffectiveDerivedRoles() {
+                    return _meta.EffectiveDerivedRoles.ToList();
+                }
+            
+                public Dictionary<string, CheckResourcesResponse.Types.ResultEntry.Types.Meta.Types.EffectMeta> GetActions() {
+                    return _meta.Actions.ToDictionary(
+                        x => x.Key,
+                        x => x.Value
+                    );
+                }
+
+                public CheckResourcesResponse.Types.ResultEntry.Types.Meta GetRaw()
+                {
+                    return _meta;
+                }
+            }
+        
+            public sealed class Outputs
+            {
+                private readonly List<OutputEntry> _outputs;
+            
+                internal Outputs(List<OutputEntry> outputs) {
+                    _outputs = outputs;
+                }
+
+                public Value Get(string src)
+                {
+                    return ToDictionary()[src];
```

Comment:
Is there a way to lazily save the result of `ToDictionary` so that subsequent calls to `Get` don't need to rebuild it?

https://github.com/cerbos/cerbos/blob/44c7afaf7c7211cc869b357dbf385e2e91575115/client/model.go#L437-L457